packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/call-http-api.ts

```diff
@@ -1,19 +1,28 @@
-import * as apigatewayv2 from '@aws-cdk/aws-apigatewayv2';
 import * as iam from '@aws-cdk/aws-iam';
 import * as sfn from '@aws-cdk/aws-stepfunctions';
 import * as cdk from '@aws-cdk/core';
-import { Construct } from 'constructs';
+import { Construct, IConstruct } from 'constructs';
 import { CallApiGatewayEndpointBase } from './base';
 import { CallApiGatewayEndpointBaseProps } from './base-types';
 
+/**
+ * An APIGateWayV2 Http API resource representation
+ */
+export interface IApiGatewayV2HttpApi extends IConstruct {
+  /**
+   * The identifier of this API Gateway HTTP API.
+   */
+  readonly apiId: string;
+}
+
 /**
  * Properties for calling an HTTP API Endpoint
  */
 export interface CallApiGatewayHttpApiEndpointProps extends CallApiGatewayEndpointBaseProps {
   /**
    * API to call
    */
-  readonly api: apigatewayv2.IHttpApi;
+  readonly api: IApiGatewayV2HttpApi;
```

Comment:
Why not just make this -

```ts
readonly apiId: string
```

Reply:
1. Because of the way it is used in the consumer. The consumer code requires the apigw containing Construct in order to get the defining Stack to get its region:
```
const apiStack = cdk.Stack.of(this.props.api);
return `${this.props.api.apiId}.execute-api.${apiStack.region}.${apiStack.urlSuffix}`;
```
The way to express it with an interface is to have it extends `IConstruct` which will enforce the requirements of a `scope`. 

Since we decided to merge this change to master (in the chat), keeping the interface has the advantage of not breaking typescript users.